requirements.txt

```diff
@@ -1,4 +1,19 @@
 pip<24.0
-immutabledict<3.0.1
-typing-extensions>=4.7,<5
-ordered-set==4.1.0
+black[jupyter]
+toml<0.11
+pytest-cov
+coverage<8.0.0
+ipykernel
+pytest<8.0
+nbval<0.11
+filecheck<0.0.24
+lit<18.0.0
+pre-commit==3.5.0
+ruff==0.1.3
+asv<0.7
+isort==5.12.0
+nbconvert>=7.7.2,<8.0.0
+# pyright version has to be fixed with `==`. The CI parses this file
+# and installs the according version for typechecking.
+pyright==1.1.333
```

Comment:
Where is the wgpu dependency? :thinking:

Reply:
It's in the `pyproject.toml`, along with riscemu

Reply:
in `[project.optional-dependencies]`